op-batcher/batcher/driver.go

```diff
@@ -272,10 +294,21 @@ func (l *BatchSubmitter) loop() {
 	// start the receipt/result processing loop
 	receiptLoopDone := make(chan struct{})
 	defer close(receiptLoopDone) // shut down receipt loop
+
+	var txpoolState atomic.Int32
+	txpoolState.Store(TxpoolGood)
 	go func() {
 		for {
 			select {
 			case r := <-receiptsCh:
+				if errors.Is(r.Err, txpool.ErrAlreadyReserved) && txpoolState.CompareAndSwap(TxpoolGood, TxpoolBlocked) {
+					l.Log.Info("incompatible tx in txpool")
+				} else if r.ID[0].chID == cancelID &&
+					txpoolState.CompareAndSwap(TxpoolCancelPending, TxpoolGood) {
+					// Set state to TxpoolGood even if the cancellation transaction ended in error
+					// since the stuck transaction could have cleared while we were waiting.
+					l.Log.Info("txpool may no longer be blocked", "err", r.Err)
```

Comment:
Just to check my understanding -- this indicates that the manager may send the cancellation tx many times if it ends in error, correct? the pool state would rotate to "Good", then if it is still not cleared, would cycle back through to blocked?

Reply:
Correct. the state might flip to Good here, but then a later publish attempt could again result in "ErrAlreadyReserved" which would kick off another cancellation attempt.